packages/venia-ui/lib/components/MiniCart/ProductList/item.js

```diff
@@ -20,52 +21,61 @@ const Item = props => {
+        closeMiniCart
     } = props;
-    const classes = mergeClasses(defaultClasses, propClasses);
 
-    const { isDeleting, removeItem } = useItem({ id, handleRemoveItem });
+    const classes = mergeClasses(defaultClasses, propClasses);
+    const itemLink = useMemo(
+        () => resourceUrl(`/${product.url_key}${product.url_suffix}`),
+        [product.url_key, product.url_suffix]
+    );
 
-    const rootClass = isDeleting ? classes.root_disabled : classes.root;
+    const { isDeleting, removeItem } = useItem({
+        id,
+        handleRemoveItem
+    });
 
-    const deleteButton = !isDeleting ? (
-        <button
-            className={classes.editButton}
-            onClick={removeItem}
-            type="button"
-        >
-            <Icon
-                size={16}
-                src={DeleteIcon}
-                classes={{ icon: classes.editIcon }}
-            />
-        </button>
-    ) : null;
+    const itemClass = isDeleting ? classes.item_disabled : classes.item;
 
     return (
-        <div className={rootClass}>
-            <Image
-                alt={product.name}
-                classes={{ root: classes.thumbnail }}
-                width={100}
-                resource={product.thumbnail.url}
-            />
-            <span className={classes.name}>{product.name}</span>
-            {deleteButton}
-            <ProductOptions
-                options={configurable_options}
-                classes={{
-                    options: classes.options
-                }}
-            />
-            <span className={classes.quantity}>{`Qty : ${quantity}`}</span>
-            <span className={classes.price}>
-                <Price
-                    currencyCode={prices.price.currency}
-                    value={prices.price.value}
+        <div className={classes.root}>
+            <Link className={itemClass} to={itemLink} onClick={closeMiniCart}>
+                <Image
+                    alt={product.name}
+                    classes={{ root: classes.thumbnail }}
+                    width={100}
+                    resource={product.thumbnail.url}
+                />
+                <span className={classes.name}>{product.name}</span>
+
+                <ProductOptions
+                    options={configurable_options}
+                    classes={{
+                        options: classes.options
+                    }}
+                />
+                <span className={classes.quantity}>{`Qty : ${quantity}`}</span>
+                <span className={classes.price}>
+                    <Price
+                        currencyCode={prices.price.currency}
+                        value={prices.price.value}
+                    />
+                    {' ea.'}
+                </span>
+            </Link>
+            <button
+                onClick={removeItem}
+                type="button"
+                className={classes.editButton}
+                disabled={isDeleting}
```

Comment:
Disabling delete button instead of removing it from the UI based on UX's review.